### PR TITLE
SlurmGCP. "All or nothing" bulk insert on requests with placements

### PR DESCRIPTION
**Pull request by mr0re1**

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py [hunk 1]
```diff
@@ -134,8 +130,14 @@ def create_instances_request(nodes, partition_name, placement_group, job_id=None
     log.debug(f"create_instances_request: {model} placement: {placement_group}")
 
     body = NSDict()
+
     body.count = len(nodes)
-    body.minCount = 1
+
+    if placement_group:
```
Comment (cdunbar13) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py:136]: Isn't it simpler to just use
```
# Only set minCount if we need placement.
if not placement_group:
     body.minCount
```

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py [hunk 2]
```diff
@@ -120,11 +120,7 @@ def per_instance_properties(node):
 
 def create_instances_request(nodes, partition_name, placement_group, job_id=None):
     """Call regionInstances.bulkInsert to create instances"""
-    assert len(nodes) > 0
-    if placement_group:
-        assert len(nodes) <= min(PLACEMENT_MAX_CNT, BULK_INSERT_LIMIT)
-    else:
-        assert len(nodes) <= BULK_INSERT_LIMIT
+    assert 0 < len(nodes) <= BULK_INSERT_LIMIT
```
Comment (cdunbar13) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py:123]: Testing the second assert later on will end up with some weird debug issues and I'm not sure what the model line will end up doing.

```
assert 0 < len(nodes) <= placement_group ? min(PLACEMENT_MAX_CNT, BULK_INSERT_LIMIT) : BULK_INSERT_LIMIT
```

**Review by cdunbar13 — APPROVED** [review]
It probably is fine but I have some comments.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py (hunk 1 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py:136]: I want to explicitly spell out "no op" branch.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py (hunk 2 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py:123]: ack

**Pull request merged**